runsc/boot/fs.go

```diff
@@ -756,6 +767,17 @@ func (c *containerMounter) createRootMount(ctx context.Context, conf *Config) (*
 	return rootInode, nil
 }
 
+// getBindMountNameAndOptions retrieves the fsName, opts, and useOverlay values
+// used for bind mounts.
+func (c *containerMounter) getBindMountNameAndOptions(conf *Config, m specs.Mount) (string, []string, bool) {
+	fd := c.fds.remove()
+	fsName := "9p"
+	opts := p9MountOptions(fd, c.getMountAccessType(m))
+	// If configured, add overlay to all writable mounts.
+	useOverlay := conf.Overlay && !mountFlags(m.Options).ReadOnly
+	return fsName, opts, useOverlay
+}
+
 // getMountNameAndOptions retrieves the fsName, opts, and useOverlay values
 // used for mounts.
 func (c *containerMounter) getMountNameAndOptions(conf *Config, m specs.Mount) (string, []string, bool, error) {
```

Comment:
I think you can contain the implementation entirely inside this function. First, check that options `bind` or `mbind` are present. If yes, you overwrite m.Type to `bind` and run the switch below as is. The rest of the code should just work.

Please add a a simple test covering this case in container_test.go.